public/js/index.js

```diff
@@ -5,13 +5,14 @@ var defaultExtraKeys = {
       cm.setOption("fullScreen", !cm.getOption("fullScreen"));
     },
     "Esc": function(cm) {
-      if (cm.getOption("fullScreen")) cm.setOption("fullScreen", false);
+       if (cm.getOption('keyMap').substr(0, 3) === 'vim') return CodeMirror.Pass;
+       else if(cm.getOption("fullScreen")) cm.setOption("fullScreen", false);
     },
     "Cmd-S": function () {
-        return CodeMirror.PASS
+        return CodeMirror.Pass
```

Comment:
Oh, you're right!
Then I have to change all my `CodeMirror.Pass` usage.
Sorry I must followed so old standard or some bad snippet.